src/ui/FacetValueSuggestions/FacetValueSuggestions.ts

```diff
@@ -0,0 +1,248 @@
+     */
+    useValueFromSearchbox: ComponentOptions.buildBooleanOption({
+      postProcessing: (value, options: IFacetValueSuggestionsOptions) => {
+        return value || !options.useQuerySuggestions;
+      }
+    }),
+
+    /**
+     * Specifies whether to display the number of results in each of the suggestions.
+     *
+     * Default value is `false`.
+     *
+     * **Note:**
+     * The number of results is an estimate.
+     *
+     * On a Standalone Search Interface, if you are redirecting on a Search Interface that has different filters,
+     *  the number of results on the Standalone Search Interface will be inaccurate.
+     *
+     * Setting this option has no effect when the `templateHelper` options is set.
+     */
+    displayEstimateNumberOfResults: ComponentOptions.buildBooleanOption({ defaultValue: false }),
+
+    /**
+     * Specifies the helper function to execute when generating suggestions shown to the end user in the
+     * [`Omnibox`]{@link Omnibox}.
+     *
+     * If not specified, a default template will be used.
+     *
+     * **Note:**
+     * > You cannot set this option directly in the component markup as an HTML attribute. You must either set it in the
+     * > [`init`]{@link init} call of your search interface (see
+     * > [Components - Passing Component Options in the init Call](https://developers.coveo.com/x/PoGfAQ#Components-PassingComponentOptionsintheinitCall)),
+     * > or before the `init` call, using the `options` top-level function (see
+     * > [Components - Passing Component Options Before the init Call](https://developers.coveo.com/x/PoGfAQ#Components-PassingComponentOptionsBeforetheinitCall)).
+     *
+     * **Example:**
+     *
+     * ```javascript
+     *
+     * var suggestionTemplate = function(row, omnibox) {
+     *   return "Searching for " + row.keyword + " in category " + row.value;
+     * };
+     *
+     * // You can set the option in the 'init' call:
+     * Coveo.init(document.querySelector("#search"), {
+     *    FacetValueSuggestions : {
+     *      templateHelper : suggestionTemplate
+     *    }
+     * });
+     *
+     * // Or before the 'init' call, using the 'options' top-level function:
+     * // Coveo.options(document.querySelector("#search"), {
+     * //   FacetValueSuggestions : {
+     * //     templateHelper : suggestionTemplate
+     * //   }
+     * // });
+     * ```
+     */
+    templateHelper: ComponentOptions.buildCustomOption<(row: IFacetValueSuggestionRow, omnibox: Omnibox) => string>(() => {
+      return null;
+    })
+  };
+
+  public fieldValueCache: SuggestionsCache<IFacetValueSuggestionRow[]> = new SuggestionsCache();
+
+  public facetValueSuggestionsProvider: IFacetValueSuggestionsProvider;
+
+  static defaultTemplate(this: FacetValueSuggestions, row: IFacetValueSuggestionRow, omnibox: Omnibox): string {
+    const keyword = DomUtils.highlightElement(row.keyword, omnibox.getText(), 'coveo-omnibox-hightlight2');
```

Comment:
Why do we need to pass in the css class for highlighting ?

Reply:
`coveo-omnibox-hightlight2` is italic highlighting, like it would in the ML query suggestions